cmd/lnd/main.go

```diff
@@ -34,9 +34,20 @@ func main() {
 	// Call the "real" main in a nested manner so the defers will properly
 	// be executed in the case of a graceful shutdown.
 	if err = lnd.Main(
-		loadedConfig, lnd.ListenerCfg{}, shutdownInterceptor,
+		loadedConfig, lnd.ListenerCfg{}, *shutdownInterceptor,
 	); err != nil {
 		_, _ = fmt.Fprintln(os.Stderr, err)
 		os.Exit(1)
 	}
+	defer func() {
+		fmt.Println("Shutdown complete")
+		err := loadedConfig.LogWriter.Close()
```

Comment:
This looks like a slight behaviour change? 

Previously: 
* Main exits with non-nil error
* defer in Main calls `loadedConfig.LogWriter.Close()` on exit
* We print error and `os.Exit(1)`

Now: 
* Main exits with non-nil error 
* We print error and `os.Exit(1)` 
-> log writer never gets closed

I'd suggest leaving the `defer` in `Main` and then just calling `shutdownInterceptor.GetExit...` here? Since it doesn't need to actually be in a defer (it's at the end anyway).